### PR TITLE
Bug 1164857 - App reviews can be sent with only blank spaces

### PR DESCRIPTION
**Pull request by tieu**

### CODE REVIEW

File: core/forms.js [hunk 1]
```diff
@@ -2,6 +2,13 @@ define('core/forms', ['core/z'], function(z) {
 
     function checkValid(form) {
         if (form) {
+            
+            if($.trim( $(form).find('textarea').val() ) == '')
```
Comment (ngokevin) [inline core/forms.js]: you can store this as a var

```var isEmpty = $.trim($(form).find('textarea').val()) == ''```

and then

```$(form).filter(':not([novalidate])').find('button[type=submit]').attr('disabled', isEmpty || !form.checkValidity());```

**Comment by tieu** [comment]
Since this only prevents the form from being submitted and didn't mark the form as invalid. Would this approach be better?

`$(form).find('textarea').val( $.trim( $(form).find('textarea').val() ) === '' ? '' : $(form).find('textarea').val() );`

**Comment by ngokevin** [comment]
maybe, i'll have to test it out. pretty hard to parse jquery blocks :)

**Comment by tieu** [comment]
Did you have a chance to test out the new changes?

File: core/forms.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ define('core/forms', ['core/z'], function(z) {
 
     function checkValid(form) {
         if (form) {
+            $(form).find('textarea').val($.trim($(form).find('textarea').val()) === '' ? '' : $(form).find('textarea').val());
```
Comment (spasovski) [inline core/forms.js]: let's put `$(form)` in a var

**Comment by spasovski** [comment]
The approach itself looks good r+wc

File: core/forms.js [hunk 3]
```diff
@@ -2,7 +2,9 @@ define('core/forms', ['core/z'], function(z) {
 
     function checkValid(form) {
         if (form) {
-            $(form).filter(':not([novalidate])').find('button[type=submit]').attr('disabled', !form.checkValidity());
+            var currentForm = $(form);
```
Comment (spasovski) [inline core/forms.js]: We tend to name jquery elements as `$currentForm` etc. I would also do another `var $textArea = $currentForm.find('textarea')` so we're not doing a jquery lookup every time.

File: core/forms.js [hunk 4]
```diff
@@ -2,7 +2,9 @@ define('core/forms', ['core/z'], function(z) {
 
     function checkValid(form) {
         if (form) {
-            $(form).filter(':not([novalidate])').find('button[type=submit]').attr('disabled', !form.checkValidity());
+            var $currentForm = $(form);
```
Comment (spasovski) [inline core/forms.js]: As I mentioned on IRC we should do `var $form = $(form), $textarea = $form.find('textarea');` and use those later.

**Comment by spasovski** [comment]
Looks good thanks
